.github/workflows/stage-2-test.yaml

```
@@ -40,6 +40,9 @@ jobs:
     steps:
       - name: "Checkout code"
         uses: actions/checkout@v4
+      - name: Install Python dependencies
+        run: |
+          pip install -r src/function_apps/foundry_relay/requirements.txt
```

Suggest making it clear that we're only testing the part of our code we really care about. 
I also think 'packages' is clearer than 'dependencies'.

```suggestion
      - name: Install Python packages for Foundry Relay only
        run: |
          pip install -r src/function_apps/foundry_relay/requirements.txt
```